.github/workflows/create-block.yml

```diff
@@ -32,6 +36,7 @@ jobs:
                   cache: npm
 
             - name: npm install, build, format and lint
+              shell: bash
               run: |
                   npm ci
-                  npm run test:create-block
+                  sh ./bin/test-create-block.sh
```

Comment:
Unless there is a GitHub Actions quirk that I am not aware of, in a normal environment this invocation is problematic, because `test-create-block.sh` isn't a POSIX shell script, but a Bash script with specific Bashisms. So it should be:

```
bash ./bin/test-create-block.sh
```

I'll open a follow-up PR.

Reply:
❤️

Reply:
(FWIW, we're [setting the shell to `bash`](https://github.com/WordPress/gutenberg/pull/38368/files#diff-b6f206a0376b45b0a060bc19682f83fcb62eec8b8fb4c2f0a16a43fd5d8e8405R39), so in practice, it shouldn't matter much IIUC.)

Reply:
> (FWIW, we're [setting the shell to `bash`](https://github.com/WordPress/gutenberg/pull/38368/files#diff-b6f206a0376b45b0a060bc19682f83fcb62eec8b8fb4c2f0a16a43fd5d8e8405R39), so in practice, it shouldn't matter much IIUC.)

Indeed, but I don't understand what that actually means, mechanically. :) That is, in the end in the runtime there must be something equivalent to [exec](https://en.wikipedia.org/wiki/Exec_(system_call)), and this could either mean that some layer is honouring the hashbang (parsing the first line of the script — `#!/bin/bash` — and thus exec'ing `/bin/bash` with the script path as argument), or that some layer is directly exec'ing entire string in the config — `bash ./bin/test-create-block.sh`. I have no idea what kind of effect `shell: bash` plays in the middle of this!

Reply:
BTW, coming back to this from another angle: Are there even that many Bashisms left? Seems like `-ne` could be actually POSIX-compliant 🤔

Reply:
Actually, the `-ne` is fine and dates back to the original Unix days, I believe. :) According to [ShellCheck](https://www.shellcheck.net/), the only Bashism is the `-e` flag passed to `echo`. Replacing `echo -e` with `printf` [should be enough](https://github.com/koalaman/shellcheck/wiki/SC3037) — just don't forget the trailing newline.

Btw, ShellCheck is available as a linter. I have it in Vim, so I'm sure it's trivial to install in VSCode. I highly recommend it. It not only prevents mistakes, but it has also taught me a lot thanks to the detailed explanations in the wiki. :)

Reply:
Thanks for the reminder! I was still copying code back and forth into [shellcheck.net](https://www.shellcheck.net/). (I have yet to try if it's clever enough to work inside of the workflow YAML 🤔 )